scripts/manage-templates.js

```diff
@@ -10,201 +10,464 @@ const __filename = fileURLToPath(import.meta.url);
+        const stat = fs.statSync(fullPath);
+
+        if (stat.isDirectory()) {
+            files.push(...getAllFiles(fullPath, baseDir));
+        } else {
+            files.push(relativePath);
+        }
+    }
+    return files;
+}
+
+// Check if two files have the same content
+function filesAreEqual(file1, file2) {
+    if (!fs.existsSync(file1) || !fs.existsSync(file2)) return false;
+
+    const stat1 = fs.statSync(file1);
+    const stat2 = fs.statSync(file2);
+
+    // Different sizes = different files
+    if (stat1.size !== stat2.size) return false;
+
+    // Compare content
+    const content1 = fs.readFileSync(file1);
+    const content2 = fs.readFileSync(file2);
+    return content1.equals(content2);
+}
+
+// Get files that are different or new compared to default
+// Returns: { filesToCopy: string[], stats: { new, modified, unchanged } }
+function getDiffFiles(sourceDir, defaultDir) {
+    const result = {
+        filesToCopy: [],
+        stats: { new: 0, modified: 0, unchanged: 0 }
+    };
+
+    if (!fs.existsSync(sourceDir)) return result;
+
+    const sourceFiles = getAllFiles(sourceDir);
+    const defaultFiles = new Set(fs.existsSync(defaultDir) ? getAllFiles(defaultDir) : []);
+
+    for (const relPath of sourceFiles) {
+        const sourcePath = path.join(sourceDir, relPath);
+        const defaultPath = path.join(defaultDir, relPath);
+
+        if (!defaultFiles.has(relPath)) {
+            // New file - not in default
+            result.filesToCopy.push(relPath);
+            result.stats.new++;
+        } else if (!filesAreEqual(sourcePath, defaultPath)) {
+            // Modified file - different from default
+            result.filesToCopy.push(relPath);
+            result.stats.modified++;
+        } else {
+            // Unchanged - same as default, don't copy
+            result.stats.unchanged++;
+        }
+    }
+
+    return result;
+}
+
+// Copy only diff files from source to target
+function copyDiffFiles(sourceDir, targetDir, filesToCopy) {
+    for (const relPath of filesToCopy) {
+        const sourcePath = path.join(sourceDir, relPath);
+        const targetPath = path.join(targetDir, relPath);
+
+        fs.ensureDirSync(path.dirname(targetPath));
+        fs.copyFileSync(sourcePath, targetPath);
+    }
+}
+
 program
   .name('template-manager')
   .description('Manage Statue SSG templates for development')
   .version('1.0.0');
 
-// LOAD: Template -> Workspace
+// Helper: Get template-specific dependencies from package.json
+// Excludes statue-ssg core dependencies
+function getTemplatePackageJson(packageJsonPath) {
+    if (!fs.existsSync(packageJsonPath)) return null;
+
+    const pkg = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+
+    // Core dependencies that should NOT be saved with template
+    const coreDeps = new Set([
+        'statue-ssg', 'marked', 'gray-matter', 'chalk', 'commander', 'fs-extra'
+    ]);
+    const coreDevDeps = new Set([
```

Comment:
we shouldn't check these in, it should be parsed/accessed automatically from the package.json